docs/proposals/nginx-extensions.md

```diff
@@ -0,0 +1,964 @@
+    - [Filters](#filters)
+    - [BackendRef](#backendref)
+    - [Policy](#policy)
+      - [Direct Policy Attachment](#direct-policy-attachment)
+      - [Inherited Policy Attachment](#inherited-policy-attachment)
+        - [Hierarchy](#hierarchy)
+      - [Challenges of Policy Attachment](#challenges-of-policy-attachment)
+  - [Prioritized NGINX Features](#prioritized-nginx-features)
+    - [High-Priority Features](#high-priority-features)
+    - [Medium-Priority Features](#medium-priority-features)
+    - [Low-Priority Features](#low-priority-features)
+    - [Features with Active Gateway API Enhancement Proposals](#features-with-active-gateway-api-enhancement-proposals)
+  - [Grouping the Features](#grouping-the-features)
+  - [API](#api)
+    - [Gateway Settings](#gateway-settings)
+      - [Future Work](#future-work)
+      - [Alternatives](#alternatives)
+    - [Response Modification](#response-modification)
+      - [Future Work](#future-work-1)
+      - [Alternatives](#alternatives-1)
+    - [TLS Settings](#tls-settings)
+      - [Future Work](#future-work-2)
+      - [Alternatives](#alternatives-2)
+    - [Client Settings](#client-settings)
+      - [Future Work](#future-work-3)
+      - [Alternatives](#alternatives-3)
+    - [Upstream Settings](#upstream-settings)
+      - [Alternatives](#alternatives-4)
+    - [Authentication](#authentication)
+      - [Future Work](#future-work-4)
+      - [Alternatives](#alternatives-5)
+    - [Observability](#observability)
+      - [Future Work](#future-work-5)
+      - [Alternatives](#alternatives-6)
+    - [Proxy Settings](#proxy-settings)
+      - [Future Work](#future-work-6)
+      - [Alternatives](#alternatives-7)
+    - [Circuit Breaker/ Backup service](#circuit-breaker-backup-service)
+      - [Alternatives](#alternatives-8)
+  - [Testing](#testing)
+  - [Security Considerations](#security-considerations)
+  - [Alternatives Considered](#alternatives-considered)
+  - [References](#references)
+
+<!-- TOC -->
+
+## Goals
+
+- Identify the set of NGINX features to deliver to users.
+- Group these features to reduce the number of APIs and improve user experience.
+- For each group, identify the type of Gateway API extension to use and the applicable Gateway API role(s).
+
+## Non-Goals
+
+- Design the API of every extension. This design work will be completed for each extension before implementation.
+- Design an API that will allow the user to insert raw NGINX config into the NGINX config that NGINX Gateway Fabric generates (e.g. [snippets in NGINX Ingress Controller](https://docs.nginx.com/nginx-ingress-controller/configuration/ingress-resources/advanced-configuration-with-snippets/)).
+
+## Gateway API Extensions
+
+The Gateway API provides many extensions that implementations can leverage to deliver features that the general-purpose API cannot address. This section provides a summary of these extensions.
+
+### GatewayClass Parameters Ref
+
+_Role(s)_: Infrastructure Provider
```

Comment:
In case of NGF, Cluster Operator controls it. More over, there is no infra provider role in case of NGF -- it's worth mentioning.

Reply:
Good point, I will update.